src/Uno.UI/UI/Xaml/Controls/ToolTip/ToolTip.cs

```diff
@@ -73,10 +129,600 @@ private void AttachToPopup()
+			}
+
+			// If the ToolTipService is opening the ToolTip, then its Placement is used,
+			// regardless of what the ToolTip.Placement has been set to.
+			var placement = m_pToolTipServicePlacementModeOverride ?? Placement;
+
+			var dimentions = new Rect(HorizontalOffset, VerticalOffset, ActualWidth, ActualHeight);
+
+			// PlacementMode.Mouse only makes sense for automatic ToolTips opened by touch or mouse.
+			if (placement == PlacementMode.Mouse
+				// UNO TODO
+				// &&
+				//(m_inputMode == AutomaticToolTipInputMode::Touch || m_inputMode == AutomaticToolTipInputMode::Mouse)
+				)
+			{
+				PerformMousePlacementWithPopup(dimentions, placement);
+			}
+			else
+			{
+				PerformNonMousePlacementWithPopup(pTargetRect, dimentions, placement);
+			}
+		}
+
+		// Sets the location of the ToolTip's Popup within the Xaml window.
+		private void PerformMousePlacementWithPopup(
+			Rect pDimentions, PlacementMode placement)
+		{
+			double tooltipActualWidth = pDimentions.Right;
+			double tooltipActualHeight = pDimentions.Bottom;
+			double horizontalOffset = pDimentions.Left;
+			double verticalOffset = pDimentions.Top;
+
+			var screenWidth = 0d;
+			var screenHeight = 0d;
+			var bIsRTL = false;
+
+			double maxX = 0.0;
+			double maxY = 0.0;
+			double left = 0.0;
+			double top = 0.0;
+			var toolTipRect = default(Rect);
+			var intersectionRect = default(Rect);
+
+			screenWidth = Windows.UI.Xaml.Window.Current.Bounds.Width;
+			screenHeight = Windows.UI.Xaml.Window.Current.Bounds.Height;
+
+			var spTarget = Target;
+
+			if (spTarget != null)
+			{
+				// UNO TODO
+				//        xaml::FlowDirection targetFlowDirection = xaml::FlowDirection_LeftToRight;
+				//spTarget->get_FlowDirection(&targetFlowDirection);
+				//bIsRTL = targetFlowDirection == xaml::FlowDirection_RightToLeft;
+				bIsRTL = false;
+
+				// We should not do placement if the target is no longer in the live tree.
+				if (!spTarget.IsLoaded)
+				{
+					return;
+				}
+			}
+
+			var spPopup = _popup;
+
+			var lastPointerEnteredPoint = CoreWindow.GetForCurrentThread().PointerPosition;
+
+			left = lastPointerEnteredPoint.X;
+			top = lastPointerEnteredPoint.Y;
+
+			// If we are in RTL mode, then flip the X coordinate around so that it appears to be in LTR mode. That
+			// means all of the LTR logic will still work.
+			if (bIsRTL)
+			{
+				left = screenWidth - left;
+			}
+
+			MovePointToPointerToolTipShowPosition(ref left, ref top, placement);
+
+			// align ToolTip with the bottom left corner of mouse bounding rectangle
+			top += m_mousePlacementVerticalOffset + verticalOffset;
+			left += horizontalOffset;
+
+			// pessimistic check of top value - can be 0 only if TextBlock().FontSize == 0
+			top = Math.Max(TOOLTIP_TOLERANCE, top);
+
+			// left can be less then TOOLTIP_tolerance if user put mouse pointer on the border of object
+			left = Math.Max(TOOLTIP_TOLERANCE, left);
+
+			maxX = screenWidth;
+			maxY = screenHeight;
+
+			toolTipRect.X = left;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['toolTipRect' is null on at least one execution path.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5404644)

Reply:
No, because it's a `struct`